VMEncryption/main/EncryptionSettingsUtil.py

```diff
@@ -209,12 +209,10 @@ def get_settings_data(self, protector_name, kv_url, kv_id, kek_url, kek_kv_id, k
             cutil.check_mhsm_id(kek_kv_id, "A ManagedHSM ID is required, but is missing or invalid.")
             cutil.check_mhsm_name(kek_kv_id, kek_url, "A ManagedHSM ID and ManagedHSM URL were provided, but their ManagedHSM names did not match")
             self._DISK_ENCRYPTION_KEY_SOURCE = CommonVariables.KeyStoreTypeManagedHSM
-        elif key_store_type:
+        elif keystoretype_flag_exists:
             raise Exception("The expected flag name and value to enable ManagedHSM is 'KeyStoreType':'{0}'. " +
```

Comment:
Looks good now . But do test the scenarios. 














Looks good to me now. But do test for all the scenarios that you could think of. Below were some of the test cases which I documented for testing my change for new parameter.


Enable with keyvault

Enable with keyvault + kek

Enable with kek and no keyvault

Enable with "KeyStoreType" set to "" - with keyvault

Enable with "KeyStoreType" set to "" - with Keyvault + kek

Enable with "KeyStoreType" set to "random" - with keyvault

Enable with "keysToReTyPe" set to "" - with keyvault

Enable with "keysToReTyPe" set to "random" - with keyvault

Enable with "keysToReTyPe" set to "ManagedHSM" - with keyvault

Enable with "keysToReTyPe" set to "manageDHsM" - with keyvault

Enable with "KeyStoreType" set to "managedhsm" - with Keyvault

Enable with "KeyStoreType" set to "managedhsm" - with Kek URL instead of managed hsm

Enable with "KeyStoreType" set to "managedhsm" - with invalid mhsm 

Enable with "KeyStoreType" set to "managedhsm" - with valid mhsm